ext/AdvancedPSLibtaskExt.jl

```diff
@@ -51,27 +80,26 @@ end
 
 # step to the next observe statement and
 # return the log probability of the transition (or nothing if done)
-function AdvancedPS.advance!(t::LibtaskTrace, isref::Bool=false)
-    isref ? AdvancedPS.load_state!(t.rng) : AdvancedPS.save_state!(t.rng)
-    AdvancedPS.inc_counter!(t.rng)
-
+function AdvancedPS.advance!(trace::LibtaskTrace, isref::Bool=false)
+    rng = get_rng(trace)
+    isref ? AdvancedPS.load_state!(rng) : AdvancedPS.save_state!(rng)
+    AdvancedPS.inc_counter!(rng)
+    set_rng!(trace, rng)
     # Move to next step
-    return Libtask.consume(t.model.ctask)
+    return Libtask.consume(trace.model.ctask)
 end
 
-# create a backward reference in task_local_storage
-function AdvancedPS.addreference!(task::Task, trace::LibtaskTrace)
-    if task.storage === nothing
-        task.storage = IdDict()
-    end
-    task.storage[:__trace] = trace
-
-    return task
+"""
+Set a backreference so that the TapedTask in `trace` stores the `trace` itself in the
+taped globals.
+"""
+function AdvancedPS.addreference!(trace::LibtaskTrace)
+    set_other_global!(trace, trace)
+    return trace
 end
 
 function AdvancedPS.update_rng!(trace::LibtaskTrace)
```

Comment:
The motivation of `update_rng!` is to obtain `rng` passed via `TapedFunction` arguments, and replace `trace.rng` with `rng. This becomes unnecessary because the new design no longer passes `rng` via args, so this function becomes a null operation. Probably okay to remove it completely.

Reply:
Removed.